packages/rest/src/__tests__/acceptance/routing/routing.acceptance.ts

```diff
@@ -135,6 +135,48 @@ describe('Routing', () => {
     );
   });
 
+  it('allows controllers to define single value array params via decorators', async () => {
+    class MyController {
+      @get('/greet')
+      greet(
+        @param.array('stringArray', 'query', {type: 'string'})
+        stringArray: string[],
+      ) {
+        return `hello ${stringArray}`;
```

Comment:
Thanks @bajtos this was extremely helpful.

Reply:
> This method does not distinguish between `stringArray` set to an array with a single item (`['hello']`) or set to a string (`'hello'`). Please update it to cover this aspect too.
> 
> For example:
> 
> The same comment applies to other new tests you are adding. It would be great to share the controller class to avoid repetition/duplication, for example using `describe` or `context`:
> 
> ```ts
> describe('array parameters', () => {
>   class MyController {
>     // ...
>   }
>  
>   it('allows controllers to define single value array params via decorators', async () => {
>     // ...
>   });
> 
>   it('allows controllers to define multi-value array params via decorators', async () => {
>     // ...
>   });
> });
> ```
> 
> In the other comment, I proposed to move the tests to `coercion.acceptance.ts`, in which case we should change test names too, e.g.
> 
> * `it('coerces a single value into an array with one item'`
> * `it('preserves array values as arrays'`

Turns out with coercion tests this is not necessary